### PR TITLE
Fix freed geometry leaving attributes in a broken state

### PR DESCRIPTION
**Pull request by davepagurek**
Resolves https://github.com/processing/p5.js/issues/6322

### Changes

This builds off of https://github.com/processing/p5.js/pull/6230 since that also refactors how enabled attributes are stored (the previous storage format of object keys causes the locations to be strings; a `Set` keeps them numbers.) We should deal with that PR and merge it first to make this one more reviewable, but for now, the diff of this PR from that one is here:  https://github.com/davepagurek/p5.js/compare/feat/faster-lines...davepagurek:p5.js:fix/free-geometry

- When applying a shader, we now explicitly disable any attributes that aren't handled by the shader, since it seems that some are being left over from the retained geometry
- When freeing geometry, there were some subtle bugs that are now fixed:
  - Previously, `hashCount` was global, but if it goes over a limit, it would try to free something from the current renderer, even if another renderer has all the cached geometry. Now, the count on the current renderer is used instead.
  - Previously, if we went over the cache limit, we just deleted the property in JS without freeing the WebGL resources, causing a slight memory leak. Now, it does both.

### Screenshots of change

Updated sketch from the issue: https://editor.p5js.org/davepagurek/sketches/znr0snuIy

Now, the fills aren't broken in the immediate mode drawing that comes after freeing the geometry:
image

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

### CODE REVIEW

**Comment by davepagurek** [comment]
Another option for this might be to disable all attributes when we're done rendering something. I haven't thought through the implications of that though, let me know if anyone has thoughts!

File: src/webgl/p5.RendererGL.Retained.js [hunk 1]
```diff
@@ -30,11 +28,9 @@ p5.RendererGL.prototype._initBufferDefaults = function(gId) {
   this._freeBuffers(gId);
 
   //@TODO remove this limit on hashes in retainedMode.geometry
-  hashCount++;
-  if (hashCount > 1000) {
+  if (Object.keys(this.retainedMode.geometry).length > 1000) {
```
Comment (aferriss) [inline src/webgl/p5.RendererGL.Retained.js:31]: Why do we impose this 1000 hash limit? Is that just an arbitrary number?

**Review by aferriss — APPROVED** [review]
lgtm, sorry I missed this one!

File: src/webgl/p5.RendererGL.Retained.js (hunk 1 above)
Comment (davepagurek) [reply src/webgl/p5.RendererGL.Retained.js:31]: I think so, kind of an arbitrary limit intended to make sure we don't run out of GPU memory. I think we could maybe be a bit smarter about this though -- I think `gl.bufferData()` throws `gl.OUT_OF_MEMORY` if we do run out?

**Pull request merged**